bingo/symbolic_regression/agraph/mutation.py

```diff
@@ -233,3 +249,172 @@ def _get_random_prune_location(individual):
             return None
         index = np.random.randint(len(indices))
         return indices[index]
+
+    def _fork_mutation(self, individual):
+        MAX_FORK_SIZE = 4
+
+        utilized_commands = individual.get_utilized_commands()
+        n_unutilized_commands = utilized_commands.count(False)
+
+        if n_unutilized_commands < 2:
```

Comment:
I don't remember where we landed on this topic, but technically we could perform the "fork" mutation with an arity 1 operator if n_unutilized_commands == 1.  Do we care enough about this corner case to change the algorithm?  I'm leaning no?

Reply:
I think we discussed the case where we only have ar1 operators in the generator, which would follow a similar pattern to a fork size of 1. It's an easy change to accommodate for a fork size of 1, but the old code also had this check. I can see tradeoffs for both options (including vs. excluding fork size = 1), but let me know which one you prefer.

Reply:
lets leave this as-is for now

Reply:
okay sounds good